### PR TITLE
Allow `Any` types for tuple except handler tests

### LINKED ISSUE #1635: tuple excepts fail when one of the classes has type `Any`
```python
from typing import Any

C = None   # type: Any
try:
    pass
except (C, Exception):
    pass
```
results in
```test.py:6: error: Exception type must be derived from BaseException```

### PR DESCRIPTION
**Pull request by pdmccormick**
Fixes #1635.

### CODE REVIEW

**Comment by pdmccormick** [comment]
@ddfisher

File: mypy/checker.py [hunk 1]
```diff
@@ -1827,13 +1827,15 @@ def visit_try_stmt(self, s: TryStmt) -> Type:
...
         type = self.accept(n)
-        if isinstance(type, AnyType):
-            return type
 
         all_types = []  # type: List[Type]
         test_types = type.items if isinstance(type, TupleType) else [type]
 
         for ttype in test_types:
+            if isinstance(ttype, AnyType):
+                all_types.append(ttype)
```
Comment (ddfisher) [inline mypy/checker.py]: You can just `return AnyType()` here instead.  `UnionType.make_simplified_union` returns `AnyType` if any of the types passed to it are `AnyType` (as it should).

Comment (ddfisher) [inline mypy/checker.py]: After looking at your tests, I take it back.  The way you're doing it here correctly catches additional non-Exception-derived types that come after an `Any`.

**Comment by ddfisher** [comment]
Looks good, thanks!

**Comment by pdmccormick** [comment]
Thanks @ddfisher!

**Pull request merged**